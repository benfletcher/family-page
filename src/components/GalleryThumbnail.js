import React from 'react';

const GalleryThumbnail = (props) => {
  // Either method here works. 'bind' is shorter, but the other is maybe more readable?
  // Leaving both until discussed.
  // const onClick = props.onClick.bind(null, props);
  const onClick = () => {
    props.onClick(props);
  };

  return (
    <div>
      <img
        alt="thumbnail"
        src={props.photoUrl}
        onClick={onClick}
      />
    </div>
  );
};

GalleryThumbnail.propTypes = {
  onClick: React.PropTypes.func.isRequired,
  photoUrl: React.PropTypes.string.isRequired,
};

export default GalleryThumbnail;
