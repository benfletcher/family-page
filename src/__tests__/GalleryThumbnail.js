/* eslint-env browser, jest */
import React from 'react';
import renderer from 'react-test-renderer';
import GalleryThumbnail from '../components/GalleryThumbnail';

test('Header snapshot is the same', () => {
  const component = renderer.create(
    <GalleryThumbnail />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
