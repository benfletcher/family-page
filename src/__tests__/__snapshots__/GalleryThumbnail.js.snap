exports[`test Header snapshot is the same 1`] = `
<div>
  <img
    alt="thumbnail"
    onClick={[Function]}
    src={undefined} />
</div>
`;
